ruby/red-arrow/test/test-record-batch-raw-records.rb

```diff
@@ -0,0 +1,401 @@
+            timestamp_usec: @timestamp_usec_array.value_data_type,
+            dict: @dict_array.value_data_type
+          )
+
+          @record_batch = Arrow::RecordBatch.new(
+            @schema, @string_values.length,
+            [
+              @string_array,
+              @uint32_array,
+              @double_array,
+              @decimal128_array,
+              @date32_array,
+              @date64_array,
+              @timestamp_sec_array,
+              @timestamp_msec_array,
+              @timestamp_usec_array,
+              @dict_array
+            ]
+          )
+
+          @expected_columnar_result = [
+            @string_values,
+            @uint32_values,
+            @double_values,
+            @decimal128_values.map {|d| d && Arrow::Decimal128.new(d) },
+            @date32_values,
+            @date64_values,
+            @timestamp_values,
+            @timestamp_values,
+            @timestamp_values,
+            @dict_indices
+          ]
+        end
+
+        test("default") do
+          @expected_columnar_result[3] = @decimal128_values.map {|x| x && BigDecimal(x) }
+          raw_records = @record_batch.raw_records
+          assert_equal(@expected_columnar_result.transpose, raw_records)
+        end
+
+        test("convert_decimal: false") do
+          raw_records = @record_batch.raw_records(convert_decimal: false)
+          assert_equal(@expected_columnar_result.transpose, raw_records)
+        end
+      end
+
+      sub_test_case("with list arrays") do
+        def setup
+          @boolean_values = [
+            [true, false],
+            nil,
+            [false, true, false, false],
+            [true],
+            [false, true, true]
+          ]
+          @boolean_list_array = Arrow::ListArray.new(
+            Arrow::ListDataType.new(name: "flags", type: :boolean),
+            @boolean_values
+          )
+          @decimal128_values = [
+            ['123.45', '234.56'],
+            ['234.56', '345.67'],
+            nil,
+            ['345.67'],
+            ['456.78', nil, '567.89']
+          ]
+          @decimal128_data_type = Arrow::Decimal128DataType.new(8, 2)
+          @decimal128_list_array = Arrow::ListArray.new(
+            Arrow::ListDataType.new(name: "decimal", data_type: @decimal128_data_type),
+            @decimal128_values
+          )
+
+          @record_batch = Arrow::RecordBatch.new(
+            Arrow::Schema.new(
+              boolean: @boolean_list_array.value_data_type,
+              decimal: @decimal128_list_array.value_data_type
+            ),
+            @boolean_values.length,
+            [
+              @boolean_list_array,
+              @decimal128_list_array
+            ]
+          )
+
+          @expected_columnar_result = [
+            @boolean_values.map {|x| x && Arrow::BooleanArray.new(x) },
+            @decimal128_values.map {|x|
+              x && Arrow::Decimal128Array.new(@decimal128_data_type, x)
+            }
```

Comment:
Do `#raw_records` may return Arrow arrays?

Reply:
`#raw_records` currently doesn't convert item arrays in a list array.